jobs/mongodb_migration/src/mongodb_migration/migrations/_20240112164500_cache_add_partial_field_in_split_descriptive_statistics.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 The HuggingFace Authors.
+
+import logging
+
+from libcommon.constants import CACHE_COLLECTION_RESPONSES, CACHE_MONGOENGINE_ALIAS
+from libcommon.simple_cache import CachedResponseDocument
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddPartialToSplitDescriptiveStatisticsCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info(
+            "If missing, add the 'partial' field with the default value None"
+            " to the cached results of split-descriptive-statistics job runner"
+        )
+        db = get_db(CACHE_MONGOENGINE_ALIAS)
+        db[CACHE_COLLECTION_RESPONSES].update_many(
+            {
+                "kind": "split-descriptive-statistics",
+                "http_status": 200,
+                "content.partial": {"$exists": False},
+            },
+            {
+                "$set": {
+                    "content.partial": False,
```

Comment:
oh no i realized that it's actually not always the case :(( some partial datasets worked before this PR - what's not working is when parquet files in parquet export are copied from the main branch and there are too many of them. For example, [CATMuS/medieval](https://huggingface.co/datasets/CATMuS/medieval/tree/refs%2Fconvert%2Fparquet/default/train) is not working (this PR aims to fix the latter case) but partial [OpenOrca](https://huggingface.co/datasets/Open-Orca/OpenOrca/tree/refs%2Fconvert%2Fparquet/default/partial-train) works. I don't know if it's easy to construct the rule that would account for these cases correctly here in migration somehow

Reply:
so basically i need to put `partial: True` for splits with 200 HTTP code in stats and `partial-` in directory name (so for example, as far as i understand, when `kind: "config-parquet" & content.partial: True`), something like this...